pkg/controller/vitesscell/reconcile_vtgate.go

```diff
@@ -128,6 +128,7 @@ func (r *ReconcileVitessCell) reconcileVtgate(ctx context.Context, vtc *planetsc
 		ExtraEnv:          vtc.Spec.Gateway.ExtraEnv,
 		ExtraVolumes:      vtc.Spec.Gateway.ExtraVolumes,
 		ExtraVolumeMounts: vtc.Spec.Gateway.ExtraVolumeMounts,
+		Annotations:       vtc.Spec.Gateway.Annotations,
```

Comment:
We already have a `PodAnnotations` field. I think we should just combine them at this call site, since the `operator/vtgate` package shouldn't need to think of these as different use cases.

```go
annotations := ...
update.Annotations(&annotations, vtc.Spec.Gateway.Annotations)
```